### ISSUE TITLE
Response when invalid endpoint is called be agent responsive

### ISSUE THREAD

**Issue by lordofthejars** [quarkusio/quarkus#41969]
### Description

Response when invalid endpoint is called be agent responsive, when calling an endpoint that doesn't exists, Quarkus returns a big HTML page with the current available endpoints which is great, but if you are curling you get a lot of HTML code that is hard to follow

### Implementation ideas

The idea would be make things agent aware, so if you detect it is a real browser then show that HTML page, but if it is curl/wget/ or any other property that shows it is a terminal, then switch to a text version.

**Comment by geoand** [comment]
I can't say I understand the problem described here. Can you describe with more detail and / or examples?

**Comment by lordofthejars** [comment]
Quarkus returns an HTML page when an endpoint doesn't exists, this HTML is huge as it has images embedded in Base64.  So it gets a really long text content, hard for a human to understand without proper rendering. So the idea would be to adapt the output of the content depending on the agent.

The content it is that big that even terminal scroll is consumed so you cannot even scroll up to the initial of the document.

To reproduce is easy, just scaffold a Quarkus project, use `quarkus dev` and finally do a curl to `localhost:8080/bye` and you'll get that big HTML page.

**Comment by geoand** [comment]
We do have what you describe, but it is based on the `Accept` (which is the proper way to do content negotiation), not the `User Agent`. 

If you do:

```bash
curl -H 'Accept: application/json' http://localhost:8080/test
```

you get something like:

```json
{
  "Resource Endpoints" : [ {
    "uri" : "/hello",
    "description" : "GET (produces:text/plain;charset=UTF-8)"
  }, {
    "uri" : "/hello/greeting/{name}",
    "description" : "GET (produces:text/plain;charset=UTF-8)"
  } ],
  "Static resources" : [ {
    "uri" : "/index.html",
    "description" : ""
  } ],
  "Additional endpoints" : [ {
    "uri" : "/q/arc",
    "description" : "CDI Overview"
  }, {
    "uri" : "/q/arc/beans",
    "description" : "Active CDI Beans"
  }, {
    "uri" : "/q/arc/observers",
    "description" : "Active CDI Observers"
  }, {
    "uri" : "/q/arc/removed-beans",
    "description" : "Removed CDI Beans"
  }, {
    "uri" : "/q/dev-ui",
    "description" : "Dev UI"
  } ]
}
```

I suppose we could do something for `text/plain` too, but I don't think it would bring any additional value.

**Comment by lordofthejars** [comment]
Ok, I understand what you mean of course, but as a developer I did only a `curl` to a method that returns a text plain, and basically I got the error, so of course I could do what you say, so I do a `curl`, I see all the chunk output and then I say:

* Oh look I know Quarkus I can pass the accept to JSON and I will get similar information but in JSON

or

* Oh what is this chunk of info, ok I don't understand anything, let me either try to read the HTML code (this is what I did) or let me open in a browser the query (which is not always possible/easy)

But in my opinion to improve the user experience, there should be something more automatic, to get what developer expects out of the box.

But it is my opinion, and I don't know if it is possible or not.

**Comment by geoand** [comment]
> But in my opinion to improve the user experience, there should be something more automatic, to get what developer expects out of the box

But what would that be when you are hitting the wrong endpoint?

**Comment by lordofthejars** [comment]
Now when I hit a wrong endpoint I got a big HTML file, I think it will be better either JSON or a Text with something like:

```
Endpoint blablabla not found. Possible endpoints:

* /hello
* /bye
```

**Comment by lordofthejars** [comment]
I mean it is not about the format per se, it could be HTML file too but at least something small that a human being can understand/parse :P

**Comment by geoand** [comment]
Ok thanks. 

Implementation wise this is trivial, but I'll leave it to @phillip-kruger to decide whether or not it's something we want to do

**Comment by phillip-kruger** [comment]
That should be doable. I'll look at it in the morning
